evennia/contrib/puzzles.py

```diff
@@ -0,0 +1,532 @@
+"""
+Puzzles System - Provides a typeclass and commands for
+objects that can be combined (i.e. 'use'd) to produce
+new objects.
+
+Evennia contribution - Henddher 2018
+
+A Puzzle is a recipe of what objects (aka parts) must
+be combined by a player so a new set of objects
+(aka results) are automatically created.
+
+Consider this simple Puzzle:
+
+    orange, mango, yogurt, blender = fruit smoothie
+
+As a Builder:
```

Comment:
I find this way of creating objects to build the recipe only to then delete them to be pretty cumbersome. I discuss the tag-alternative in answer to your question below.

Reply:
Not quite ... see below